src/providers/postgres/qgspostgresprovider.cpp

```diff
@@ -4259,6 +4259,44 @@ static QVariant parseStringArray( const QString &txt )
   return result;
 }
 
+static QVariant parseMultidimensionalArray( const QString &txt )
+{
+  QStringList result;
+  if ( !txt.startsWith( '{' ) || !txt.endsWith( '}' ) )
+  {
+    QgsLogger::warning( "Error parsing array, missing curly braces: " + txt );
```

Comment:
Should that be sent to `QgsMessageLog` or via `pushError`? Right now it will only be visible for devs and not for sysadmins.

Reply:
I keep the same logger method that other methods that parse string (getNextString, parseHStore, parseStringArray). I can replace all QgsLogger with pushError in file, but I'm not sure when we have to use QgsLogger, QgsMessageLog or pushError.

Reply:
`QgsLogger` is for messages for debugging / developers (i.e. you suspect they are triggered by some broken code)
`QgsMessageLog` is for messages to investigate issues by sysadmins (e.g. to help a sysadmin to fix configurations)
`pushError` is for users and grabs immediate attention if it happens. I think providers should not often require to directly talk to the user.

Reply:
Thank you for the clarification of the different log levels

> I think providers should not often require to directly talk to the user.

So, If I understand well, QgsMessageLog seems to be the more appropriate debug level for all parsing methods. If OK, I can change all the parsing methods call to QgsLogger.

Reply:
I made the modification 8f7c4e0

Reply:
@m-kuhn Is the last modification 8f7c4e0 is OK to you ? 
Is there anything else that prevent this PR from being merged ?